### PR TITLE
Update model card for gpt neox japanese

### PR DESCRIPTION
**Pull request by ahnjj**
# What does this PR do?
Update the model card gpt neox japanese for  https://github.com/huggingface/transformers/issues/36979

## Before submitting

- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 1]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
 rendered properly in your Markdown viewer.
 
 -->
+<div style="float: right;">
+    <div class="flex flex-wrap space-x-1">
+           <img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColorF=white">
+<img alt="FlasFhAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]: GPT-NeoX-Japanese doesn't support FlashAttention yet

```suggestion
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 2]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
+    </div>
+</div>
 
 # GPT-NeoX-Japanese
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
-</div>
+GPT-NeoX-Japanese, a Japanese language model based on [GPT-NeoX](https://github.com/EleutherAI/gpt-neox).
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
GPT-NeoX-Japanese, a Japanese language model based on [GPT-NeoX](./gpt_neox).
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 3]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
+Japanese is pretty special—it uses three types of characters (hiragana, katakana, kanji) and has a huge vocabulary.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
Japanese uses three types of characters (hiragana, katakana, kanji) and has a huge vocabulary. This model uses [BPEEncoder V2](https://github.com/tanreinama/Japanese-BPEEncoder_V2), a sub-word tokenizer to handle the different characters.
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 4]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
 
-## Overview
+To handle that, we used a special [special sub-word tokenizer](https://github.com/tanreinama/Japanese-BPEEncoder_V2) made by *tanreinama*. Huge thanks to them for open-sourcing this!
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 5]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
-Japanese is a unique language with its large vocabulary and a combination of hiragana, katakana, and kanji writing scripts.
-To address this distinct structure of the Japanese language, we use a [special sub-word tokenizer](https://github.com/tanreinama/Japanese-BPEEncoder_V2). We are very grateful to *tanreinama* for open-sourcing this incredibly helpful tokenizer.
-Following the recommendations from Google's research on [PaLM](https://ai.googleblog.com/2022/04/pathways-language-model-palm-scaling-to.html), we have removed bias parameters from transformer blocks, achieving better model performance. Please refer [this article](https://medium.com/ml-abeja/training-a-better-gpt-2-93b157662ae4) in detail.
+We also followed tips from Google’s [PaLM](https://ai.googleblog.com/2022/04/pathways-language-model-palm-scaling-to.html) research and removed some bias parameters in the model to help it perform better. Please refer [this article](https://medium.com/ml-abeja/training-a-better-gpt-2-93b157662ae4) in detail.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
The model also removes some bias parameters for better performance.
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 6]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
 
-Development of the model was led by [Shinya Otani](https://github.com/SO0529), [Takayoshi Makabe](https://github.com/spider-man-tm), [Anuj Arora](https://github.com/Anuj040), and [Kyo Hattori](https://github.com/go5paopao) from [ABEJA, Inc.](https://www.abejainc.com/). For more information on this model-building activity, please refer [here (ja)](https://tech-blog.abeja.asia/entry/abeja-gpt-project-202207).
+You can find all the original [GPT-NeoX-Japanese] checkpoints under the [gpt-neox](https://github.com/EleutherAI/gpt-neox) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
You can find all the original GPT-NeoX-Japanese checkpoints under the [ABEJA](https://huggingface.co/abeja/models?search=gpt-neo-x) organization.
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 7]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
 
-### Usage example
+> [!TIP]
+> This model was contributed by [Shinya Otani](https://github.com/SO0529), [Takayoshi Makabe](https://github.com/spider-man-tm), [Anuj Arora](https://github.com/Anuj040), and [Kyo Hattori](https://github.com/go5paopao) from [ABEJA, Inc.](https://www.abejainc.com/).
+>
+> Click on the GPT-NeoX-Japanese models in the right sidebar for more examples of how to apply GPT-NeoX-Japanese to different language tasks.
 
-The `generate()` method can be used to generate text using GPT NeoX Japanese model.
+The example below demonstrates how to [insert task here] with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
The example below demonstrates how to generate text with [`Pipeline`] or the [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 8]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
+pipeline("人とAIが協調するためには、")
+```
 
->>> prompt = "人とAIが協調するためには、"
+</hfoption>
+<hfoption id="AutoModel">
 
->>> input_ids = tokenizer(prompt, return_tensors="pt").input_ids
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```py
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer

device = "cuda" if torch.cuda.is_available() else "cpu"
model = AutoModelForCausalLM.from_pretrained("abeja/gpt-neox-japanese-2.7b", torch_dtype=torch.float16, device_map="auto").to(device)
tokenizer = AutoTokenizer.from_pretrained("abeja/gpt-neox-japanese-2.7b")
input_ids = tokenizer("人とAIが協調するためには、", return_tensors="pt").input_ids.to(device)
outputs = model.generate(input_ids)
print(tokenizer.decode(outputs[0], skip_special_tokens=True))
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 9]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
 
->>> print(gen_text)
-人とAIが協調するためには、AIと人が共存し、AIを正しく理解する必要があります。
+input_ids = tokenizer("人とAIが協調するためには、", return_tensors="pt").input_ids.to(device)
+outputs = model.generate(input_ids)
+print(tokenizer.decode(outputs[0], skip_special_tokens=True))
 ```
 
+</hfoption>
+<hfoption id="transformers-cli">
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
<hfoption id="transformers CLI">
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 10]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
+
+```bash
+echo -e "人とAIが協調するためには、" | transformers-cli run --task text-generation --model abeja/gpt-neox-japanese-2.7b --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]:
```suggestion
echo -e "人とAIが協調するためには、" | transformers run --task text-generation --model abeja/gpt-neox-japanese-2.7b --device 0
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 11]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
+```
+
+Use the [AttentionMaskVisualizer](https://github.com/huggingface/transformers/blob/beb9b5b02246b9b7ee81ddf938f93f44cfeaad19/src/transformers/utils/attention_visualizer.py#L139) to better understand what tokens the model can and cannot attend to.
+
+\```py
+from transformers.utils.attention_visualizer import AttentionMaskVisualizer
+
+visualizer = AttentionMaskVisualizer("abeja/gpt-neox-japanese-2.7b")
+visualizer("<img>What is shown in this image?")
+\```
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md]: You can remove the `\`, I was only using that to escape the backticks :)

```suggestion
```py
from transformers.utils.attention_visualizer import AttentionMaskVisualizer

visualizer = AttentionMaskVisualizer("abeja/gpt-neox-japanese-2.7b")
visualizer("<img>What is shown in this image?")
```
```

File: docs/source/en/model_doc/gpt_neox_japanese.md [hunk 12]
```diff
@@ -13,49 +13,107 @@ specific language governing permissions and limitations under the License.
...
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/gpt_neox_japanese-attn-mask.png"/>
+</div>
+
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/gpt_neox_japanese.md:117]: We can remove the task guide below and instead replace it with:

```md
Refer to the [Training a better GPT model: Learnings from PaLM](https://medium.com/ml-abeja/training-a-better-gpt-2-93b157662ae4) blog post for more details about how ABEJA trained GPT-NeoX-Japanese.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for your contribution!

**Comment by ahnjj** [comment]
@stevhliu Thanks so much for all the details, and sorry for the late response. 
I’ve updated the amendments accordingly. 🤗

**Review by stevhliu — APPROVED** [review]
LGTM, thanks again!
